core/src/acc_text/nlg/gf/grammar.clj

```diff
@@ -47,6 +47,18 @@
       (format "Item. x%s ::= \"%s\";" (name id) value)
       (format "ItemMod%d. x%s ::= \"%s\" %s;" (count relations) (name id) value (join-relation-ids relations) (su/escape-string value)))))
 
+(defmethod build-fragment :sequence [{id ::sg/id relations ::sg/relations}]
+  (when (seq relations)
+    (format "Sequence%d. x%s ::= %s;" (count relations) (name id) (join-relation-ids relations))))
+
+(defmethod build-fragment :shuffle [{id ::sg/id relations ::sg/relations}]
+  (when (seq relations)
+    (format "Sequence%d. x%s ::= %s;" (count relations) (name id) (join-relation-ids (shuffle relations)))))
```

Comment:
It is not relevant while we're working only on single row, but in general we want random sequence for each row, not get single variation and apply to all rows

Reply:
Maybe adding each permutation as a separate row (while keeping same label and Symbol) and leaving for GF to choose randomly, is a solution